share/p2p/shrex/client.go

```diff
@@ -0,0 +1,172 @@
+package shrex
+
+import (
+	"errors"
+	"fmt"
+
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"golang.org/x/net/context"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/ipld"
+	share_p2p_v1 "github.com/celestiaorg/celestia-node/share/p2p/v1/pb"
+	"github.com/celestiaorg/go-libp2p-messenger/serde"
+	"github.com/celestiaorg/nmt/namespace"
+)
+
+var (
+	errInvalidStatusCode = errors.New("INVALID")
+	errNotFound          = errors.New("NOT_FOUND")
+	errServerInternal    = errors.New("SERVER_INTERNAL")
+	errConnRefused       = errors.New("CONN_REFUSED")
```

Comment:
I dont think we need to create an error variable for each of these, can we not just do an errors.New(status.String()) in the switch and have the cases `fallthrough`?

Reply:
Errors has been planned to be handled to punish broken peers (errInvalidStatusCode, errServerInternal). Removed for now, since no peer management is implemented in this RP